In [1]:
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error
from xgboost import XGBRegressor
from dataset_test import make_dataset
from os.path import join


train_path = join('jeju_data', 'train_new.parquet')
test_path = join('jeju_data', 'test_new.parquet')

x_train, y_train, test = make_dataset(train_path, test_path)

Start time:  2022-10-22 14:39:44.581731
Train dataset success !
Test dataset success !
End time: 2022-10-22 14:40:55.710094
Play time:  0:01:11.128363


In [2]:
X = x_train.copy()
y = y_train.copy()

# x_train, x_val, y_train, y_val = train_test_split(x_train, y_train, test_size=0.2, shuffle=True, random_state=1103)

In [3]:
import optuna
from optuna import Trial
from optuna.samplers import TPESampler
from catboost import CatBoostRegressor

In [23]:

def objective(trial):
  param = {
      "random_state":42,
      'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
      'bagging_temperature' :trial.suggest_loguniform('bagging_temperature', 0.01, 100.00),
      "n_estimators":trial.suggest_int("n_estimators", 500, 5000),
      "max_depth":trial.suggest_int("max_depth", 4, 16),
      'random_strength' :trial.suggest_int('random_strength', 0, 100),
      "colsample_bylevel":trial.suggest_float("colsample_bylevel", 0.4, 1.0),
      "l2_leaf_reg":trial.suggest_float("l2_leaf_reg",1e-8,3e-5),
      "min_child_samples": trial.suggest_int("min_child_samples", 5, 100),
      "max_bin": trial.suggest_int("max_bin", 200, 500),
      'od_type': trial.suggest_categorical('od_type', ['IncToDec', 'Iter']),
  }
  X_train, X_valid, y_train, y_valid = train_test_split(X,y,test_size=0.3)
  cat = CatBoostRegressor(**param)
  cat.fit(X_train, y_train,
          eval_set=[(X_train, y_train), (X_valid,y_valid)],
          early_stopping_rounds=50,
          verbose=100)
  cat_pred = cat.predict(X_valid)
  mae = mean_absolute_error(y_valid, cat_pred)

  return mae

sampler = TPESampler(seed=42)
study = optuna.create_study(
    study_name = 'cat_parameter_opt',
    direction = 'minimize',
    sampler = sampler,
)

study.optimize(objective, n_trials=30)
print("Best Score:",study.best_value)
print("Best trial",study.best_trial.params)

[I 2022-10-23 13:41:26,058] A new study created in memory with name: cat_parameter_opt
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'bagging_temperature' :trial.suggest_loguniform('bagging_temperature', 0.01, 100.00),


0:	learn: 15.7712344	test: 15.7712344	test1: 15.7743621	best: 15.7743621 (0)	total: 289ms	remaining: 18m 15s
100:	learn: 8.4444179	test: 8.4444179	test1: 8.4399513	best: 8.4399513 (100)	total: 27.2s	remaining: 16m 34s
200:	learn: 7.3270225	test: 7.3270225	test1: 7.3235087	best: 7.3235087 (200)	total: 53.2s	remaining: 15m 50s
300:	learn: 6.9431849	test: 6.9431849	test1: 6.9400495	best: 6.9400495 (300)	total: 1m 18s	remaining: 15m 13s
400:	learn: 6.6838840	test: 6.6838840	test1: 6.6812799	best: 6.6812799 (400)	total: 1m 44s	remaining: 14m 40s
500:	learn: 6.5009679	test: 6.5009679	test1: 6.4994151	best: 6.4994151 (500)	total: 2m 8s	remaining: 14m 6s
600:	learn: 6.3534288	test: 6.3534288	test1: 6.3531122	best: 6.3531122 (600)	total: 2m 34s	remaining: 13m 38s
700:	learn: 6.2323504	test: 6.2323504	test1: 6.2326925	best: 6.2326925 (700)	total: 2m 59s	remaining: 13m 11s
800:	learn: 6.1205720	test: 6.1205720	test1: 6.1217662	best: 6.1217662 (800)	total: 3m 25s	remaining: 12m 46s
900:	learn: 5.9

[I 2022-10-23 13:57:32,359] Trial 0 finished with value: 3.376312378072731 and parameters: {'learning_rate': 0.018272261776066247, 'bagging_temperature': 63.512210106407046, 'n_estimators': 3794, 'max_depth': 11, 'random_strength': 15, 'colsample_bylevel': 0.49359671220172163, 'l2_leaf_reg': 1.7519275289243016e-06, 'min_child_samples': 88, 'max_bin': 380, 'od_type': 'IncToDec'}. Best is trial 0 with value: 3.376312378072731.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/re

0:	learn: 15.5672418	test: 15.5672418	test1: 15.5758501	best: 15.5758501 (0)	total: 187ms	remaining: 4m 32s
100:	learn: 8.5149793	test: 8.5149793	test1: 8.5057943	best: 8.5057943 (100)	total: 16.9s	remaining: 3m 45s
200:	learn: 7.7970269	test: 7.7970269	test1: 7.7872818	best: 7.7872818 (200)	total: 33.2s	remaining: 3m 26s
300:	learn: 7.3836334	test: 7.3836334	test1: 7.3740719	best: 7.3740719 (300)	total: 49.5s	remaining: 3m 9s
400:	learn: 6.9966669	test: 6.9966669	test1: 6.9866298	best: 6.9866298 (400)	total: 1m 5s	remaining: 2m 52s
500:	learn: 6.6969027	test: 6.6969027	test1: 6.6879304	best: 6.6879304 (500)	total: 1m 21s	remaining: 2m 35s
600:	learn: 6.4836264	test: 6.4836264	test1: 6.4746684	best: 6.4746684 (600)	total: 1m 38s	remaining: 2m 19s
700:	learn: 6.3204340	test: 6.3204340	test1: 6.3120120	best: 6.3120120 (700)	total: 1m 54s	remaining: 2m 2s
800:	learn: 6.1728511	test: 6.1728511	test1: 6.1641541	best: 6.1641541 (800)	total: 2m 10s	remaining: 1m 46s
900:	learn: 6.0451528	test

[I 2022-10-23 14:01:32,417] Trial 1 finished with value: 4.136473098245934 and parameters: {'learning_rate': 0.04763628595029446, 'bagging_temperature': 21.368329072358772, 'n_estimators': 1455, 'max_depth': 6, 'random_strength': 18, 'colsample_bylevel': 0.5825453457757226, 'l2_leaf_reg': 1.5747445384650815e-05, 'min_child_samples': 46, 'max_bin': 287, 'od_type': 'IncToDec'}. Best is trial 0 with value: 3.376312378072731.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/relea

0:	learn: 15.7767621	test: 15.7767621	test1: 15.7667051	best: 15.7667051 (0)	total: 521ms	remaining: 22m 8s
100:	learn: 8.2483904	test: 8.2483904	test1: 8.2424891	best: 8.2424891 (100)	total: 54s	remaining: 21m 51s
200:	learn: 7.0082720	test: 7.0082720	test1: 7.0031847	best: 7.0031847 (200)	total: 1m 49s	remaining: 21m 20s
300:	learn: 6.6093389	test: 6.6093389	test1: 6.6060506	best: 6.6060506 (300)	total: 2m 42s	remaining: 20m 18s
400:	learn: 6.3704181	test: 6.3704181	test1: 6.3691949	best: 6.3691949 (400)	total: 3m 35s	remaining: 19m 18s
500:	learn: 6.2076127	test: 6.2076127	test1: 6.2084021	best: 6.2084021 (500)	total: 4m 26s	remaining: 18m 12s
600:	learn: 6.0655183	test: 6.0655183	test1: 6.0676936	best: 6.0676936 (600)	total: 5m 17s	remaining: 17m 10s
700:	learn: 5.9607836	test: 5.9607836	test1: 5.9646395	best: 5.9646395 (700)	total: 6m 7s	remaining: 16m 10s
800:	learn: 5.8699927	test: 5.8699927	test1: 5.8755754	best: 5.8755754 (800)	total: 6m 56s	remaining: 15m 11s
900:	learn: 5.78

[I 2022-10-23 14:24:16,294] Trial 2 finished with value: 3.4188697055487456 and parameters: {'learning_rate': 0.016002960978292496, 'bagging_temperature': 0.2920433847181412, 'n_estimators': 2552, 'max_depth': 14, 'random_strength': 20, 'colsample_bylevel': 0.708540663048167, 'l2_leaf_reg': 1.7776512920172654e-05, 'min_child_samples': 9, 'max_bin': 382, 'od_type': 'IncToDec'}. Best is trial 0 with value: 3.376312378072731.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/rele

0:	learn: 15.5323292	test: 15.5323292	test1: 15.5277123	best: 15.5277123 (0)	total: 185ms	remaining: 12m 44s
100:	learn: 8.0816818	test: 8.0816818	test1: 8.0790249	best: 8.0790249 (100)	total: 18.1s	remaining: 12m 2s
200:	learn: 7.3071888	test: 7.3071888	test1: 7.3055034	best: 7.3055034 (200)	total: 35.8s	remaining: 11m 41s
300:	learn: 6.8712266	test: 6.8712266	test1: 6.8694858	best: 6.8694858 (300)	total: 53.5s	remaining: 11m 21s
400:	learn: 6.5434367	test: 6.5434367	test1: 6.5409296	best: 6.5409296 (400)	total: 1m 11s	remaining: 11m 1s
500:	learn: 6.3034066	test: 6.3034066	test1: 6.2999514	best: 6.2999514 (500)	total: 1m 28s	remaining: 10m 44s
600:	learn: 6.1079986	test: 6.1079986	test1: 6.1043932	best: 6.1043932 (600)	total: 1m 46s	remaining: 10m 24s
700:	learn: 5.9504272	test: 5.9504272	test1: 5.9472446	best: 5.9472446 (700)	total: 2m 3s	remaining: 10m 6s
800:	learn: 5.8208051	test: 5.8208051	test1: 5.8177795	best: 5.8177795 (800)	total: 2m 21s	remaining: 9m 49s
900:	learn: 5.71103

[I 2022-10-23 14:36:29,533] Trial 3 finished with value: 3.3646428031879094 and parameters: {'learning_rate': 0.04605136717611768, 'bagging_temperature': 72.86653737491046, 'n_estimators': 4138, 'max_depth': 7, 'random_strength': 9, 'colsample_bylevel': 0.8105398159072941, 'l2_leaf_reg': 1.3210173287250643e-05, 'min_child_samples': 16, 'max_bin': 349, 'od_type': 'Iter'}. Best is trial 3 with value: 3.3646428031879094.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/

0:	learn: 15.8350772	test: 15.8350772	test1: 15.8434270	best: 15.8434270 (0)	total: 207ms	remaining: 6m 33s
100:	learn: 9.7301686	test: 9.7301686	test1: 9.7260739	best: 9.7260739 (100)	total: 21.9s	remaining: 6m 30s
200:	learn: 8.4349348	test: 8.4349348	test1: 8.4295322	best: 8.4295322 (200)	total: 43.6s	remaining: 6m 9s
300:	learn: 7.9729654	test: 7.9729654	test1: 7.9682396	best: 7.9682396 (300)	total: 1m 5s	remaining: 5m 47s
400:	learn: 7.6696883	test: 7.6696883	test1: 7.6660147	best: 7.6660147 (400)	total: 1m 26s	remaining: 5m 22s
500:	learn: 7.4858414	test: 7.4858414	test1: 7.4832800	best: 7.4832800 (500)	total: 1m 46s	remaining: 4m 58s
600:	learn: 7.3309445	test: 7.3309445	test1: 7.3291655	best: 7.3291655 (600)	total: 2m 7s	remaining: 4m 36s
700:	learn: 7.1993867	test: 7.1993867	test1: 7.1983456	best: 7.1983456 (700)	total: 2m 28s	remaining: 4m 14s
800:	learn: 7.0781711	test: 7.0781711	test1: 7.0777342	best: 7.0777342 (800)	total: 2m 49s	remaining: 3m 53s
900:	learn: 6.9712526	tes

[I 2022-10-23 14:43:18,373] Trial 4 finished with value: 4.2689958893610305 and parameters: {'learning_rate': 0.015166293102182283, 'bagging_temperature': 4.467752817973908, 'n_estimators': 1903, 'max_depth': 10, 'random_strength': 55, 'colsample_bylevel': 0.5109126733153162, 'l2_leaf_reg': 2.9087842986659113e-05, 'min_child_samples': 79, 'max_bin': 482, 'od_type': 'IncToDec'}. Best is trial 3 with value: 3.3646428031879094.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/re

0:	learn: 15.6564632	test: 15.6564632	test1: 15.6692361	best: 15.6692361 (0)	total: 131ms	remaining: 3m 1s
100:	learn: 9.5496704	test: 9.5496704	test1: 9.5491533	best: 9.5491533 (100)	total: 13.6s	remaining: 2m 52s
200:	learn: 8.8608132	test: 8.8608132	test1: 8.8609534	best: 8.8609534 (200)	total: 26.9s	remaining: 2m 37s
300:	learn: 8.4650243	test: 8.4650243	test1: 8.4658989	best: 8.4658989 (300)	total: 40.1s	remaining: 2m 24s
400:	learn: 8.0559881	test: 8.0559881	test1: 8.0575386	best: 8.0575386 (400)	total: 53.4s	remaining: 2m 10s
500:	learn: 7.7119770	test: 7.7119770	test1: 7.7132895	best: 7.7132895 (500)	total: 1m 6s	remaining: 1m 57s
600:	learn: 7.4704498	test: 7.4704498	test1: 7.4718907	best: 7.4718907 (600)	total: 1m 19s	remaining: 1m 43s
700:	learn: 7.2719186	test: 7.2719186	test1: 7.2733188	best: 7.2733188 (700)	total: 1m 33s	remaining: 1m 30s
800:	learn: 7.0933077	test: 7.0933077	test1: 7.0944879	best: 7.0944879 (800)	total: 1m 46s	remaining: 1m 17s
900:	learn: 6.9492249	test

[I 2022-10-23 14:46:24,775] Trial 5 finished with value: 4.902467515077717 and parameters: {'learning_rate': 0.04409226795827594, 'bagging_temperature': 0.022592797420156956, 'n_estimators': 1382, 'max_depth': 4, 'random_strength': 32, 'colsample_bylevel': 0.6332063738136893, 'l2_leaf_reg': 8.147757462899138e-06, 'min_child_samples': 84, 'max_bin': 307, 'od_type': 'Iter'}. Best is trial 3 with value: 3.3646428031879094.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/release

0:	learn: 15.8407583	test: 15.8407583	test1: 15.8348224	best: 15.8348224 (0)	total: 1.2s	remaining: 16m 42s
100:	learn: 9.8537355	test: 9.8537355	test1: 9.8498500	best: 9.8498500 (100)	total: 1m 14s	remaining: 9m 1s
200:	learn: 8.1306359	test: 8.1306359	test1: 8.1289627	best: 8.1289627 (200)	total: 2m 42s	remaining: 8m 33s
300:	learn: 7.5500235	test: 7.5500235	test1: 7.5500957	best: 7.5500957 (300)	total: 4m 4s	remaining: 7m 13s
400:	learn: 7.2097198	test: 7.2097198	test1: 7.2112803	best: 7.2112803 (400)	total: 5m 20s	remaining: 5m 46s
500:	learn: 7.0166198	test: 7.0166198	test1: 7.0196568	best: 7.0196568 (500)	total: 6m 33s	remaining: 4m 22s
600:	learn: 6.8451546	test: 6.8451546	test1: 6.8499539	best: 6.8499539 (600)	total: 7m 51s	remaining: 3m 3s
700:	learn: 6.7295899	test: 6.7295899	test1: 6.7358083	best: 6.7358083 (700)	total: 9m 4s	remaining: 1m 44s
800:	learn: 6.6518858	test: 6.6518858	test1: 6.6589422	best: 6.6589422 (800)	total: 10m 1s	remaining: 25.5s
834:	learn: 6.6294179	tes

[I 2022-10-23 14:56:54,470] Trial 6 finished with value: 5.04283461084448 and parameters: {'learning_rate': 0.012545899554294089, 'bagging_temperature': 16.172900811143155, 'n_estimators': 835, 'max_depth': 16, 'random_strength': 77, 'colsample_bylevel': 0.5192294089205034, 'l2_leaf_reg': 1.7560829253683595e-07, 'min_child_samples': 83, 'max_bin': 412, 'od_type': 'Iter'}. Best is trial 3 with value: 3.3646428031879094.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases

0:	learn: 15.8331530	test: 15.8331530	test1: 15.8506284	best: 15.8506284 (0)	total: 476ms	remaining: 8m 5s
100:	learn: 9.8786878	test: 9.8786878	test1: 9.8995465	best: 9.8995465 (100)	total: 57.4s	remaining: 8m 43s
200:	learn: 8.1202121	test: 8.1202121	test1: 8.1419332	best: 8.1419332 (200)	total: 1m 53s	remaining: 7m 42s
300:	learn: 7.5259987	test: 7.5259987	test1: 7.5481214	best: 7.5481214 (300)	total: 2m 44s	remaining: 6m 34s
400:	learn: 7.2254681	test: 7.2254681	test1: 7.2482571	best: 7.2482571 (400)	total: 3m 34s	remaining: 5m 31s
500:	learn: 7.0026605	test: 7.0026605	test1: 7.0261133	best: 7.0261133 (500)	total: 4m 27s	remaining: 4m 37s
600:	learn: 6.8325391	test: 6.8325391	test1: 6.8569759	best: 6.8569759 (600)	total: 5m 18s	remaining: 3m 42s
700:	learn: 6.7098218	test: 6.7098218	test1: 6.7349921	best: 6.7349921 (700)	total: 6m 6s	remaining: 2m 47s
800:	learn: 6.6083948	test: 6.6083948	test1: 6.6342260	best: 6.6342260 (800)	total: 6m 57s	remaining: 1m 54s
900:	learn: 6.5303433	t

[I 2022-10-23 15:05:45,798] Trial 7 finished with value: 4.8975912235202985 and parameters: {'learning_rate': 0.011265617213006592, 'bagging_temperature': 0.27155819552829413, 'n_estimators': 1021, 'max_depth': 15, 'random_strength': 62, 'colsample_bylevel': 0.5985388149115896, 'l2_leaf_reg': 1.9161149250778487e-06, 'min_child_samples': 34, 'max_bin': 297, 'od_type': 'IncToDec'}. Best is trial 3 with value: 3.3646428031879094.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/

0:	learn: 15.6640044	test: 15.6640044	test1: 15.6528718	best: 15.6528718 (0)	total: 133ms	remaining: 2m 18s
100:	learn: 7.5922719	test: 7.5922719	test1: 7.5883570	best: 7.5883570 (100)	total: 37.3s	remaining: 5m 45s
200:	learn: 7.0043822	test: 7.0043822	test1: 7.0056154	best: 7.0056154 (200)	total: 1m 12s	remaining: 5m 1s
300:	learn: 6.6852421	test: 6.6852421	test1: 6.6893416	best: 6.6893416 (300)	total: 1m 44s	remaining: 4m 16s
400:	learn: 6.3315855	test: 6.3315855	test1: 6.3390817	best: 6.3390817 (400)	total: 2m 21s	remaining: 3m 44s
500:	learn: 5.8236729	test: 5.8236729	test1: 5.8345877	best: 5.8345877 (500)	total: 3m	remaining: 3m 13s
600:	learn: 5.4969695	test: 5.4969695	test1: 5.5102728	best: 5.5102728 (600)	total: 3m 39s	remaining: 2m 39s
700:	learn: 5.2769837	test: 5.2769837	test1: 5.2927377	best: 5.2927377 (700)	total: 4m 19s	remaining: 2m 4s
800:	learn: 5.1200863	test: 5.1200863	test1: 5.1380642	best: 5.1380642 (800)	total: 4m 59s	remaining: 1m 28s
900:	learn: 5.0040667	test:

[I 2022-10-23 15:12:26,299] Trial 8 finished with value: 3.5473664753038716 and parameters: {'learning_rate': 0.04169990777997927, 'bagging_temperature': 0.7742116473996251, 'n_estimators': 1038, 'max_depth': 13, 'random_strength': 76, 'colsample_bylevel': 0.7367663185416977, 'l2_leaf_reg': 2.3131305726837285e-05, 'min_child_samples': 52, 'max_bin': 357, 'od_type': 'IncToDec'}. Best is trial 3 with value: 3.3646428031879094.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/re

0:	learn: 15.8580828	test: 15.8580828	test1: 15.8639143	best: 15.8639143 (0)	total: 194ms	remaining: 10m 51s
100:	learn: 10.5458360	test: 10.5458360	test1: 10.5553345	best: 10.5553345 (100)	total: 19.5s	remaining: 10m 29s
200:	learn: 9.1162127	test: 9.1162127	test1: 9.1290395	best: 9.1290395 (200)	total: 38.5s	remaining: 10m 5s
300:	learn: 8.5773716	test: 8.5773716	test1: 8.5916924	best: 8.5916924 (300)	total: 57.3s	remaining: 9m 42s
400:	learn: 8.2263373	test: 8.2263373	test1: 8.2414550	best: 8.2414550 (400)	total: 1m 16s	remaining: 9m 22s
500:	learn: 7.9973535	test: 7.9973535	test1: 8.0122855	best: 8.0122855 (500)	total: 1m 34s	remaining: 9m 2s
600:	learn: 7.8220202	test: 7.8220202	test1: 7.8366853	best: 7.8366853 (600)	total: 1m 53s	remaining: 8m 39s
700:	learn: 7.6791412	test: 7.6791412	test1: 7.6935786	best: 7.6935786 (700)	total: 2m 11s	remaining: 8m 20s
800:	learn: 7.5525196	test: 7.5525196	test1: 7.5668223	best: 7.5668223 (800)	total: 2m 30s	remaining: 8m 1s
900:	learn: 7.44883

[I 2022-10-23 15:23:00,437] Trial 9 finished with value: 4.203360745142629 and parameters: {'learning_rate': 0.011896326393433528, 'bagging_temperature': 0.0133572404119741, 'n_estimators': 3364, 'max_depth': 8, 'random_strength': 51, 'colsample_bylevel': 0.9445398843556558, 'l2_leaf_reg': 7.486273952174759e-06, 'min_child_samples': 44, 'max_bin': 427, 'od_type': 'IncToDec'}. Best is trial 3 with value: 3.3646428031879094.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/rele

0:	learn: 15.6835512	test: 15.6835512	test1: 15.6946777	best: 15.6946777 (0)	total: 186ms	remaining: 15m 2s
100:	learn: 8.6004585	test: 8.6004585	test1: 8.6005617	best: 8.6005617 (100)	total: 17.9s	remaining: 14m 5s
200:	learn: 7.6785880	test: 7.6785880	test1: 7.6797049	best: 7.6797049 (200)	total: 35.3s	remaining: 13m 38s
300:	learn: 7.2363044	test: 7.2363044	test1: 7.2380231	best: 7.2380231 (300)	total: 52.5s	remaining: 13m 15s
400:	learn: 6.9274924	test: 6.9274924	test1: 6.9282307	best: 6.9282307 (400)	total: 1m 9s	remaining: 12m 56s
500:	learn: 6.6847250	test: 6.6847250	test1: 6.6845402	best: 6.6845402 (500)	total: 1m 27s	remaining: 12m 37s
600:	learn: 6.5041987	test: 6.5041987	test1: 6.5036014	best: 6.5036014 (600)	total: 1m 44s	remaining: 12m 19s
700:	learn: 6.3509409	test: 6.3509409	test1: 6.3495387	best: 6.3495387 (700)	total: 2m 1s	remaining: 12m 2s
800:	learn: 6.2191867	test: 6.2191867	test1: 6.2173824	best: 6.2173824 (800)	total: 2m 18s	remaining: 11m 44s
900:	learn: 6.09607

[I 2022-10-23 15:37:05,824] Trial 10 finished with value: 3.481128480161777 and parameters: {'learning_rate': 0.02881012453203432, 'bagging_temperature': 5.335218224577685, 'n_estimators': 4863, 'max_depth': 7, 'random_strength': 1, 'colsample_bylevel': 0.8727055573922243, 'l2_leaf_reg': 1.0845746773289858e-05, 'min_child_samples': 6, 'max_bin': 229, 'od_type': 'Iter'}. Best is trial 3 with value: 3.3646428031879094.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/t

0:	learn: 15.6789560	test: 15.6789560	test1: 15.6873738	best: 15.6873738 (0)	total: 240ms	remaining: 16m 22s
100:	learn: 7.7514592	test: 7.7514592	test1: 7.7618330	best: 7.7618330 (100)	total: 24.5s	remaining: 16m 8s
200:	learn: 6.9276551	test: 6.9276551	test1: 6.9402422	best: 6.9402422 (200)	total: 48.2s	remaining: 15m 32s
300:	learn: 6.5604326	test: 6.5604326	test1: 6.5730692	best: 6.5730692 (300)	total: 1m 11s	remaining: 15m 2s
400:	learn: 6.2976071	test: 6.2976071	test1: 6.3105991	best: 6.3105991 (400)	total: 1m 35s	remaining: 14m 35s
500:	learn: 6.0825009	test: 6.0825009	test1: 6.0958246	best: 6.0958246 (500)	total: 1m 58s	remaining: 14m 11s
600:	learn: 5.9080792	test: 5.9080792	test1: 5.9216649	best: 5.9216649 (600)	total: 2m 22s	remaining: 13m 47s
700:	learn: 5.7530287	test: 5.7530287	test1: 5.7664927	best: 5.7664927 (700)	total: 2m 45s	remaining: 13m 21s
800:	learn: 5.6268168	test: 5.6268168	test1: 5.6407553	best: 5.6407553 (800)	total: 3m 9s	remaining: 12m 58s
900:	learn: 5.52

[I 2022-10-23 15:53:20,816] Trial 11 finished with value: 3.2693161826057278 and parameters: {'learning_rate': 0.026337960285887585, 'bagging_temperature': 73.50694050446648, 'n_estimators': 4093, 'max_depth': 10, 'random_strength': 2, 'colsample_bylevel': 0.802945466237456, 'l2_leaf_reg': 4.810014901266662e-06, 'min_child_samples': 98, 'max_bin': 466, 'od_type': 'Iter'}. Best is trial 11 with value: 3.2693161826057278.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/release

0:	learn: 15.6594912	test: 15.6594912	test1: 15.6543025	best: 15.6543025 (0)	total: 279ms	remaining: 21m 36s
100:	learn: 7.6073531	test: 7.6073531	test1: 7.6122138	best: 7.6122138 (100)	total: 25s	remaining: 18m 45s
200:	learn: 6.8413549	test: 6.8413549	test1: 6.8454581	best: 6.8454581 (200)	total: 49.1s	remaining: 18m 6s
300:	learn: 6.4577632	test: 6.4577632	test1: 6.4612463	best: 6.4612463 (300)	total: 1m 13s	remaining: 17m 36s
400:	learn: 6.1782581	test: 6.1782581	test1: 6.1811776	best: 6.1811776 (400)	total: 1m 37s	remaining: 17m 11s
500:	learn: 5.9678715	test: 5.9678715	test1: 5.9706420	best: 5.9706420 (500)	total: 2m 1s	remaining: 16m 45s
600:	learn: 5.7932701	test: 5.7932701	test1: 5.7960930	best: 5.7960930 (600)	total: 2m 25s	remaining: 16m 21s
700:	learn: 5.6455011	test: 5.6455011	test1: 5.6484270	best: 5.6484270 (700)	total: 2m 49s	remaining: 15m 56s
800:	learn: 5.5249634	test: 5.5249634	test1: 5.5278820	best: 5.5278820 (800)	total: 3m 14s	remaining: 15m 33s
900:	learn: 5.424

[I 2022-10-23 16:12:38,104] Trial 12 finished with value: 3.2084640981675783 and parameters: {'learning_rate': 0.028604187862629173, 'bagging_temperature': 98.87000462348665, 'n_estimators': 4653, 'max_depth': 10, 'random_strength': 1, 'colsample_bylevel': 0.8342132377544605, 'l2_leaf_reg': 1.1467284672749309e-05, 'min_child_samples': 100, 'max_bin': 495, 'od_type': 'Iter'}. Best is trial 12 with value: 3.2084640981675783.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/rele

0:	learn: 15.6922100	test: 15.6922100	test1: 15.6957683	best: 15.6957683 (0)	total: 325ms	remaining: 25m 46s
100:	learn: 7.9294303	test: 7.9294303	test1: 7.9346738	best: 7.9346738 (100)	total: 32.6s	remaining: 25m 2s
200:	learn: 7.1281170	test: 7.1281170	test1: 7.1323228	best: 7.1323228 (200)	total: 1m 4s	remaining: 24m 15s
300:	learn: 6.8259554	test: 6.8259554	test1: 6.8294038	best: 6.8294038 (300)	total: 1m 35s	remaining: 23m 27s
400:	learn: 6.5960328	test: 6.5960328	test1: 6.5997589	best: 6.5997589 (400)	total: 2m 6s	remaining: 22m 50s
500:	learn: 6.4143589	test: 6.4143589	test1: 6.4181074	best: 6.4181074 (500)	total: 2m 36s	remaining: 22m 10s
600:	learn: 6.2369527	test: 6.2369527	test1: 6.2407884	best: 6.2407884 (600)	total: 3m 7s	remaining: 21m 37s
700:	learn: 6.0035610	test: 6.0035610	test1: 6.0065137	best: 6.0065137 (700)	total: 3m 38s	remaining: 21m 4s
800:	learn: 5.7960228	test: 5.7960228	test1: 5.7989180	best: 5.7989180 (800)	total: 4m 9s	remaining: 20m 33s
900:	learn: 5.6335

[I 2022-10-23 16:38:11,877] Trial 13 finished with value: 3.17501550397241 and parameters: {'learning_rate': 0.026817901075278292, 'bagging_temperature': 87.90612496943417, 'n_estimators': 4755, 'max_depth': 11, 'random_strength': 33, 'colsample_bylevel': 0.9980387183177496, 'l2_leaf_reg': 6.716486964548631e-06, 'min_child_samples': 97, 'max_bin': 499, 'od_type': 'Iter'}. Best is trial 13 with value: 3.17501550397241.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/

0:	learn: 15.6531988	test: 15.6531988	test1: 15.6489844	best: 15.6489844 (0)	total: 393ms	remaining: 32m 8s
100:	learn: 7.6217243	test: 7.6217243	test1: 7.6250677	best: 7.6250677 (100)	total: 40.1s	remaining: 31m 47s
200:	learn: 6.9265130	test: 6.9265130	test1: 6.9337877	best: 6.9337877 (200)	total: 1m 18s	remaining: 30m 27s
300:	learn: 6.6150576	test: 6.6150576	test1: 6.6250933	best: 6.6250933 (300)	total: 1m 56s	remaining: 29m 40s
400:	learn: 6.3979238	test: 6.3979238	test1: 6.4096818	best: 6.4096818 (400)	total: 2m 33s	remaining: 28m 47s
500:	learn: 6.2017351	test: 6.2017351	test1: 6.2145456	best: 6.2145456 (500)	total: 3m 10s	remaining: 27m 59s
600:	learn: 5.9382891	test: 5.9382891	test1: 5.9513174	best: 5.9513174 (600)	total: 3m 50s	remaining: 27m 33s
700:	learn: 5.6918608	test: 5.6918608	test1: 5.7043200	best: 5.7043200 (700)	total: 4m 30s	remaining: 27m 1s
800:	learn: 5.5128890	test: 5.5128890	test1: 5.5250447	best: 5.5250447 (800)	total: 5m 9s	remaining: 26m 25s
900:	learn: 5.3

[I 2022-10-23 17:12:05,189] Trial 14 finished with value: 3.110995629227421 and parameters: {'learning_rate': 0.029883904534194723, 'bagging_temperature': 10.88881792577849, 'n_estimators': 4909, 'max_depth': 12, 'random_strength': 34, 'colsample_bylevel': 0.9872068331087231, 'l2_leaf_reg': 1.9856488736372744e-05, 'min_child_samples': 69, 'max_bin': 485, 'od_type': 'Iter'}. Best is trial 14 with value: 3.110995629227421.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releas

0:	learn: 15.6139282	test: 15.6139282	test1: 15.6142911	best: 15.6142911 (0)	total: 375ms	remaining: 31m 12s
100:	learn: 7.4847081	test: 7.4847081	test1: 7.4849938	best: 7.4849938 (100)	total: 38.5s	remaining: 31m 3s
200:	learn: 6.8470565	test: 6.8470565	test1: 6.8482743	best: 6.8482743 (200)	total: 1m 16s	remaining: 30m 31s
300:	learn: 6.5395464	test: 6.5395464	test1: 6.5423172	best: 6.5423172 (300)	total: 1m 52s	remaining: 29m 14s
400:	learn: 6.3136478	test: 6.3136478	test1: 6.3180324	best: 6.3180324 (400)	total: 2m 29s	remaining: 28m 36s
500:	learn: 6.0409296	test: 6.0409296	test1: 6.0477009	best: 6.0477009 (500)	total: 3m 7s	remaining: 27m 59s
600:	learn: 5.7400527	test: 5.7400527	test1: 5.7481789	best: 5.7481789 (600)	total: 3m 45s	remaining: 27m 26s
700:	learn: 5.5082849	test: 5.5082849	test1: 5.5177389	best: 5.5177389 (700)	total: 4m 23s	remaining: 26m 56s
800:	learn: 5.3476996	test: 5.3476996	test1: 5.3597305	best: 5.3597305 (800)	total: 5m 2s	remaining: 26m 26s
900:	learn: 5.2

[I 2022-10-23 17:45:37,693] Trial 15 finished with value: 3.0831404491932544 and parameters: {'learning_rate': 0.03379605099283127, 'bagging_temperature': 9.983703827612532, 'n_estimators': 4995, 'max_depth': 12, 'random_strength': 35, 'colsample_bylevel': 0.9790965241670061, 'l2_leaf_reg': 2.1653946012786313e-05, 'min_child_samples': 68, 'max_bin': 453, 'od_type': 'Iter'}. Best is trial 15 with value: 3.0831404491932544.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/relea

0:	learn: 15.6153051	test: 15.6153051	test1: 15.5974105	best: 15.5974105 (0)	total: 483ms	remaining: 26m 27s
100:	learn: 7.4921933	test: 7.4921933	test1: 7.4873970	best: 7.4873970 (100)	total: 45.8s	remaining: 24m 6s
200:	learn: 6.8347480	test: 6.8347480	test1: 6.8341752	best: 6.8341752 (200)	total: 1m 29s	remaining: 23m 2s
300:	learn: 6.5382077	test: 6.5382077	test1: 6.5403746	best: 6.5403746 (300)	total: 2m 14s	remaining: 22m 13s
400:	learn: 6.3158746	test: 6.3158746	test1: 6.3202068	best: 6.3202068 (400)	total: 2m 55s	remaining: 21m 3s
500:	learn: 5.9937970	test: 5.9937970	test1: 5.9999719	best: 5.9999719 (500)	total: 3m 41s	remaining: 20m 32s
600:	learn: 5.6594359	test: 5.6594359	test1: 5.6663898	best: 5.6663898 (600)	total: 4m 28s	remaining: 20m
700:	learn: 5.4107850	test: 5.4107850	test1: 5.4196643	best: 5.4196643 (700)	total: 5m 16s	remaining: 19m 29s
800:	learn: 5.2362803	test: 5.2362803	test1: 5.2471381	best: 5.2471381 (800)	total: 6m 5s	remaining: 18m 55s
900:	learn: 5.108007

[I 2022-10-23 18:13:07,777] Trial 16 finished with value: 3.1228287721870918 and parameters: {'learning_rate': 0.03459483714551643, 'bagging_temperature': 3.4239459861409047, 'n_estimators': 3291, 'max_depth': 13, 'random_strength': 44, 'colsample_bylevel': 0.9590900920882006, 'l2_leaf_reg': 2.1291913407597456e-05, 'min_child_samples': 66, 'max_bin': 439, 'od_type': 'Iter'}. Best is trial 15 with value: 3.0831404491932544.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/rele

0:	learn: 15.6134798	test: 15.6134798	test1: 15.6328368	best: 15.6328368 (0)	total: 369ms	remaining: 30m 44s
100:	learn: 7.4195905	test: 7.4195905	test1: 7.4313213	best: 7.4313213 (100)	total: 36.1s	remaining: 29m 9s
200:	learn: 6.8005708	test: 6.8005708	test1: 6.8125074	best: 6.8125074 (200)	total: 1m 10s	remaining: 28m 7s
300:	learn: 6.4798174	test: 6.4798174	test1: 6.4930117	best: 6.4930117 (300)	total: 1m 44s	remaining: 27m 6s
400:	learn: 6.2408165	test: 6.2408165	test1: 6.2557089	best: 6.2557089 (400)	total: 2m 18s	remaining: 26m 28s
500:	learn: 5.9135223	test: 5.9135223	test1: 5.9284617	best: 5.9284617 (500)	total: 2m 53s	remaining: 25m 58s
600:	learn: 5.6329112	test: 5.6329112	test1: 5.6482617	best: 5.6482617 (600)	total: 3m 28s	remaining: 25m 28s
700:	learn: 5.4235823	test: 5.4235823	test1: 5.4394729	best: 5.4394729 (700)	total: 4m 4s	remaining: 24m 56s
800:	learn: 5.2683912	test: 5.2683912	test1: 5.2854247	best: 5.2854247 (800)	total: 4m 40s	remaining: 24m 28s
900:	learn: 5.14

[I 2022-10-23 18:45:01,926] Trial 17 finished with value: 3.0729984933759353 and parameters: {'learning_rate': 0.03580731155673778, 'bagging_temperature': 14.835342351104112, 'n_estimators': 4997, 'max_depth': 12, 'random_strength': 34, 'colsample_bylevel': 0.890804407282392, 'l2_leaf_reg': 2.755132801878939e-05, 'min_child_samples': 67, 'max_bin': 446, 'od_type': 'Iter'}. Best is trial 17 with value: 3.0729984933759353.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releas

0:	learn: 15.6149060	test: 15.6149060	test1: 15.6243471	best: 15.6243471 (0)	total: 629ms	remaining: 28m 13s
100:	learn: 7.8813717	test: 7.8813717	test1: 7.9060732	best: 7.9060732 (100)	total: 54.5s	remaining: 23m 16s
200:	learn: 7.0994543	test: 7.0994543	test1: 7.1264365	best: 7.1264365 (200)	total: 1m 47s	remaining: 22m 8s
300:	learn: 6.7558260	test: 6.7558260	test1: 6.7849240	best: 6.7849240 (300)	total: 2m 37s	remaining: 20m 47s
400:	learn: 6.5789754	test: 6.5789754	test1: 6.6094701	best: 6.6094701 (400)	total: 3m 20s	remaining: 19m 6s
500:	learn: 6.1557185	test: 6.1557185	test1: 6.1888916	best: 6.1888916 (500)	total: 4m 19s	remaining: 18m 54s
600:	learn: 5.7024839	test: 5.7024839	test1: 5.7353163	best: 5.7353163 (600)	total: 5m 23s	remaining: 18m 44s
700:	learn: 5.4031365	test: 5.4031365	test1: 5.4350866	best: 5.4350866 (700)	total: 6m 28s	remaining: 18m 21s
800:	learn: 5.2067945	test: 5.2067945	test1: 5.2388202	best: 5.2388202 (800)	total: 7m 33s	remaining: 17m 50s
900:	learn: 5.

[I 2022-10-23 19:14:02,513] Trial 18 finished with value: 3.1338460665129944 and parameters: {'learning_rate': 0.03544226791393884, 'bagging_temperature': 1.305034269179579, 'n_estimators': 2691, 'max_depth': 14, 'random_strength': 100, 'colsample_bylevel': 0.8972587408398297, 'l2_leaf_reg': 2.6897337910524628e-05, 'min_child_samples': 65, 'max_bin': 449, 'od_type': 'Iter'}. Best is trial 17 with value: 3.0729984933759353.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/rele

0:	learn: 15.7956562	test: 15.7956562	test1: 15.7961973	best: 15.7961973 (0)	total: 186ms	remaining: 13m 31s
100:	learn: 8.2280408	test: 8.2280408	test1: 8.2266179	best: 8.2266179 (100)	total: 35.4s	remaining: 24m 56s
200:	learn: 7.2547618	test: 7.2547618	test1: 7.2532390	best: 7.2532390 (200)	total: 1m 11s	remaining: 24m 49s
300:	learn: 6.9044294	test: 6.9044294	test1: 6.9044763	best: 6.9044763 (300)	total: 1m 46s	remaining: 24m
400:	learn: 6.6955975	test: 6.6955975	test1: 6.6965255	best: 6.6965255 (400)	total: 2m 19s	remaining: 23m 3s
500:	learn: 6.5385872	test: 6.5385872	test1: 6.5412087	best: 6.5412087 (500)	total: 2m 54s	remaining: 22m 29s
600:	learn: 6.3916011	test: 6.3916011	test1: 6.3958228	best: 6.3958228 (600)	total: 3m 28s	remaining: 21m 48s
700:	learn: 6.2401948	test: 6.2401948	test1: 6.2461335	best: 6.2461335 (700)	total: 4m 3s	remaining: 21m 15s
800:	learn: 6.0179624	test: 6.0179624	test1: 6.0251897	best: 6.0251897 (800)	total: 4m 39s	remaining: 20m 44s
900:	learn: 5.8064

[I 2022-10-23 19:41:19,923] Trial 19 finished with value: 3.1979959627220587 and parameters: {'learning_rate': 0.022199040331204678, 'bagging_temperature': 1.5331324344171082, 'n_estimators': 4367, 'max_depth': 12, 'random_strength': 41, 'colsample_bylevel': 0.9086990631339946, 'l2_leaf_reg': 2.459261658611784e-05, 'min_child_samples': 31, 'max_bin': 399, 'od_type': 'Iter'}. Best is trial 17 with value: 3.0729984933759353.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/rele

0:	learn: 15.7470204	test: 15.7470204	test1: 15.7538650	best: 15.7538650 (0)	total: 246ms	remaining: 14m 8s
100:	learn: 8.7063602	test: 8.7063602	test1: 8.7022567	best: 8.7022567 (100)	total: 23.1s	remaining: 12m 46s
200:	learn: 7.7687757	test: 7.7687757	test1: 7.7655890	best: 7.7655890 (200)	total: 45.7s	remaining: 12m 17s
300:	learn: 7.3192580	test: 7.3192580	test1: 7.3162325	best: 7.3162325 (300)	total: 1m 7s	remaining: 11m 49s
400:	learn: 7.0667953	test: 7.0667953	test1: 7.0642770	best: 7.0642770 (400)	total: 1m 29s	remaining: 11m 23s
500:	learn: 6.8874898	test: 6.8874898	test1: 6.8854202	best: 6.8854202 (500)	total: 1m 52s	remaining: 10m 59s
600:	learn: 6.7324059	test: 6.7324059	test1: 6.7306473	best: 6.7306473 (600)	total: 2m 14s	remaining: 10m 36s
700:	learn: 6.5599064	test: 6.5599064	test1: 6.5584574	best: 6.5584574 (700)	total: 2m 37s	remaining: 10m 15s
800:	learn: 6.3640854	test: 6.3640854	test1: 6.3628672	best: 6.3628672 (800)	total: 2m 59s	remaining: 9m 51s
900:	learn: 6.18

[I 2022-10-23 19:53:58,864] Trial 20 finished with value: 3.52715505352058 and parameters: {'learning_rate': 0.02201472430013196, 'bagging_temperature': 20.34682434829078, 'n_estimators': 3446, 'max_depth': 9, 'random_strength': 26, 'colsample_bylevel': 0.7446304262717984, 'l2_leaf_reg': 2.893601055648774e-05, 'min_child_samples': 59, 'max_bin': 335, 'od_type': 'Iter'}. Best is trial 17 with value: 3.0729984933759353.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/

0:	learn: 15.6339489	test: 15.6339489	test1: 15.6391279	best: 15.6391279 (0)	total: 404ms	remaining: 33m 28s
100:	learn: 7.5800793	test: 7.5800793	test1: 7.5765771	best: 7.5765771 (100)	total: 38.2s	remaining: 30m 44s
200:	learn: 6.8804970	test: 6.8804970	test1: 6.8784886	best: 6.8784886 (200)	total: 1m 15s	remaining: 30m
300:	learn: 6.5728595	test: 6.5728595	test1: 6.5719750	best: 6.5719750 (300)	total: 1m 52s	remaining: 29m 4s
400:	learn: 6.3589433	test: 6.3589433	test1: 6.3597607	best: 6.3597607 (400)	total: 2m 29s	remaining: 28m 21s
500:	learn: 6.1122344	test: 6.1122344	test1: 6.1141691	best: 6.1141691 (500)	total: 3m 6s	remaining: 27m 48s
600:	learn: 5.8012680	test: 5.8012680	test1: 5.8035796	best: 5.8035796 (600)	total: 3m 45s	remaining: 27m 18s
700:	learn: 5.5699214	test: 5.5699214	test1: 5.5728614	best: 5.5728614 (700)	total: 4m 23s	remaining: 26m 45s
800:	learn: 5.3957428	test: 5.3957428	test1: 5.4000239	best: 5.4000239 (800)	total: 5m 1s	remaining: 26m 11s
900:	learn: 5.25621

[I 2022-10-23 20:27:45,113] Trial 21 finished with value: 3.088109336272345 and parameters: {'learning_rate': 0.033139084315120665, 'bagging_temperature': 9.825485200681358, 'n_estimators': 4972, 'max_depth': 12, 'random_strength': 37, 'colsample_bylevel': 0.998481897088584, 'l2_leaf_reg': 1.9928914096140035e-05, 'min_child_samples': 72, 'max_bin': 467, 'od_type': 'Iter'}. Best is trial 17 with value: 3.0729984933759353.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releas

0:	learn: 15.6036289	test: 15.6036289	test1: 15.6100524	best: 15.6100524 (0)	total: 479ms	remaining: 35m 23s
100:	learn: 7.2895076	test: 7.2895076	test1: 7.2999177	best: 7.2999177 (100)	total: 46.8s	remaining: 33m 28s
200:	learn: 6.6865939	test: 6.6865939	test1: 6.6958392	best: 6.6958392 (200)	total: 1m 30s	remaining: 31m 37s
300:	learn: 6.4339580	test: 6.4339580	test1: 6.4447341	best: 6.4447341 (300)	total: 2m 11s	remaining: 30m 10s
400:	learn: 6.2290812	test: 6.2290812	test1: 6.2408224	best: 6.2408224 (400)	total: 2m 53s	remaining: 29m 8s
500:	learn: 5.8945859	test: 5.8945859	test1: 5.9067931	best: 5.9067931 (500)	total: 3m 39s	remaining: 28m 42s
600:	learn: 5.5858770	test: 5.5858770	test1: 5.5980393	best: 5.5980393 (600)	total: 4m 27s	remaining: 28m 24s
700:	learn: 5.3537023	test: 5.3537023	test1: 5.3662621	best: 5.3662621 (700)	total: 5m 15s	remaining: 28m 2s
800:	learn: 5.1871056	test: 5.1871056	test1: 5.2012149	best: 5.2012149 (800)	total: 6m 4s	remaining: 27m 37s
900:	learn: 5.0

[I 2022-10-23 21:05:57,253] Trial 22 finished with value: 3.0581063861881272 and parameters: {'learning_rate': 0.036326827652300754, 'bagging_temperature': 7.941853727983881, 'n_estimators': 4438, 'max_depth': 13, 'random_strength': 42, 'colsample_bylevel': 0.9288894494388297, 'l2_leaf_reg': 2.5373559102164518e-05, 'min_child_samples': 72, 'max_bin': 455, 'od_type': 'Iter'}. Best is trial 22 with value: 3.0581063861881272.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/rele

0:	learn: 15.5824808	test: 15.5824808	test1: 15.5739079	best: 15.5739079 (0)	total: 624ms	remaining: 45m 42s
100:	learn: 7.4051905	test: 7.4051905	test1: 7.4044055	best: 7.4044055 (100)	total: 54.9s	remaining: 38m 55s
200:	learn: 6.7832606	test: 6.7832606	test1: 6.7876600	best: 6.7876600 (200)	total: 1m 51s	remaining: 38m 37s
300:	learn: 6.4919401	test: 6.4919401	test1: 6.5000258	best: 6.5000258 (300)	total: 2m 41s	remaining: 36m 38s
400:	learn: 6.2248887	test: 6.2248887	test1: 6.2357970	best: 6.2357970 (400)	total: 3m 33s	remaining: 35m 28s
500:	learn: 5.7619949	test: 5.7619949	test1: 5.7726573	best: 5.7726573 (500)	total: 4m 33s	remaining: 35m 29s
600:	learn: 5.4262379	test: 5.4262379	test1: 5.4376299	best: 5.4376299 (600)	total: 5m 35s	remaining: 35m 19s
700:	learn: 5.2128267	test: 5.2128267	test1: 5.2256424	best: 5.2256424 (700)	total: 6m 39s	remaining: 35m 4s
800:	learn: 5.0506948	test: 5.0506948	test1: 5.0659170	best: 5.0659170 (800)	total: 7m 41s	remaining: 34m 32s
900:	learn: 4

[I 2022-10-23 21:54:52,710] Trial 23 finished with value: 3.0243654977100873 and parameters: {'learning_rate': 0.038335486405884085, 'bagging_temperature': 24.760294060675108, 'n_estimators': 4397, 'max_depth': 14, 'random_strength': 61, 'colsample_bylevel': 0.9205580450819787, 'l2_leaf_reg': 2.5541984632868808e-05, 'min_child_samples': 57, 'max_bin': 440, 'od_type': 'Iter'}. Best is trial 23 with value: 3.0243654977100873.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/rel

0:	learn: 15.5259187	test: 15.5259187	test1: 15.5309084	best: 15.5309084 (0)	total: 1.13s	remaining: 1h 13m 47s
100:	learn: 7.2725901	test: 7.2725901	test1: 7.2816491	best: 7.2816491 (100)	total: 1m 23s	remaining: 52m 31s
200:	learn: 6.7228614	test: 6.7228614	test1: 6.7384204	best: 6.7384204 (200)	total: 2m 30s	remaining: 46m 23s
300:	learn: 6.4203923	test: 6.4203923	test1: 6.4409677	best: 6.4409677 (300)	total: 3m 36s	remaining: 43m 12s
400:	learn: 6.1070149	test: 6.1070149	test1: 6.1334331	best: 6.1334331 (400)	total: 4m 53s	remaining: 42m 48s
500:	learn: 5.5593267	test: 5.5593267	test1: 5.5887629	best: 5.5887629 (500)	total: 6m 41s	remaining: 45m 29s
600:	learn: 5.2347896	test: 5.2347896	test1: 5.2662645	best: 5.2662645 (600)	total: 8m 30s	remaining: 46m 54s
700:	learn: 5.0361317	test: 5.0361317	test1: 5.0711927	best: 5.0711927 (700)	total: 10m 21s	remaining: 47m 27s
800:	learn: 4.8887491	test: 4.8887491	test1: 4.9286640	best: 4.9286640 (800)	total: 12m 16s	remaining: 47m 38s
900:	l

[I 2022-10-23 23:10:02,339] Trial 24 finished with value: 3.0206542230860727 and parameters: {'learning_rate': 0.04053844897057419, 'bagging_temperature': 34.85340043850737, 'n_estimators': 3911, 'max_depth': 16, 'random_strength': 61, 'colsample_bylevel': 0.40015726738282775, 'l2_leaf_reg': 2.5808786872558374e-05, 'min_child_samples': 57, 'max_bin': 420, 'od_type': 'Iter'}. Best is trial 24 with value: 3.0206542230860727.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/rele

0:	learn: 15.5747593	test: 15.5747593	test1: 15.5637965	best: 15.5637965 (0)	total: 724ms	remaining: 45m 43s
100:	learn: 7.5070836	test: 7.5070836	test1: 7.5127114	best: 7.5127114 (100)	total: 1m 20s	remaining: 49m 12s
200:	learn: 6.8785306	test: 6.8785306	test1: 6.8906818	best: 6.8906818 (200)	total: 2m 25s	remaining: 43m 10s
300:	learn: 6.5291233	test: 6.5291233	test1: 6.5462572	best: 6.5462572 (300)	total: 3m 30s	remaining: 40m 40s
400:	learn: 6.2830817	test: 6.2830817	test1: 6.3048374	best: 6.3048374 (400)	total: 4m 35s	remaining: 38m 50s
500:	learn: 5.7682960	test: 5.7682960	test1: 5.7929504	best: 5.7929504 (500)	total: 6m 12s	remaining: 40m 41s
600:	learn: 5.3755959	test: 5.3755959	test1: 5.4033028	best: 5.4033028 (600)	total: 7m 58s	remaining: 42m 19s
700:	learn: 5.1260858	test: 5.1260858	test1: 5.1578513	best: 5.1578513 (700)	total: 9m 51s	remaining: 43m 24s
800:	learn: 4.9666340	test: 4.9666340	test1: 5.0027119	best: 5.0027119 (800)	total: 11m 45s	remaining: 43m 51s
900:	learn

[I 2022-10-24 00:21:47,071] Trial 25 finished with value: 3.0412939144270217 and parameters: {'learning_rate': 0.037957388889815875, 'bagging_temperature': 31.971232927558667, 'n_estimators': 3788, 'max_depth': 16, 'random_strength': 69, 'colsample_bylevel': 0.40458137603376054, 'l2_leaf_reg': 2.4986199815925864e-05, 'min_child_samples': 52, 'max_bin': 411, 'od_type': 'Iter'}. Best is trial 24 with value: 3.0206542230860727.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/re

0:	learn: 15.5408976	test: 15.5408976	test1: 15.5444974	best: 15.5444974 (0)	total: 735ms	remaining: 45m 22s
100:	learn: 7.3124039	test: 7.3124039	test1: 7.3175665	best: 7.3175665 (100)	total: 1m 26s	remaining: 51m 25s
200:	learn: 6.7485327	test: 6.7485327	test1: 6.7584219	best: 6.7584219 (200)	total: 2m 31s	remaining: 44m 4s
300:	learn: 6.4865092	test: 6.4865092	test1: 6.5001272	best: 6.5001272 (300)	total: 3m 25s	remaining: 38m 40s
400:	learn: 6.0834541	test: 6.0834541	test1: 6.1018967	best: 6.1018967 (400)	total: 4m 43s	remaining: 38m 59s
500:	learn: 5.5010702	test: 5.5010702	test1: 5.5201165	best: 5.5201165 (500)	total: 6m 34s	remaining: 42m 2s
600:	learn: 5.1850194	test: 5.1850194	test1: 5.2058473	best: 5.2058473 (600)	total: 8m 25s	remaining: 43m 32s
700:	learn: 4.9770278	test: 4.9770278	test1: 5.0017646	best: 5.0017646 (700)	total: 10m 19s	remaining: 44m 16s
800:	learn: 4.8359362	test: 4.8359362	test1: 4.8671536	best: 4.8671536 (800)	total: 12m 15s	remaining: 44m 26s
900:	learn:

[I 2022-10-24 01:34:51,731] Trial 26 finished with value: 2.996221780232261 and parameters: {'learning_rate': 0.042366009030577655, 'bagging_temperature': 34.88131118422019, 'n_estimators': 3705, 'max_depth': 16, 'random_strength': 68, 'colsample_bylevel': 0.41110669305970876, 'l2_leaf_reg': 2.4954723999222694e-05, 'min_child_samples': 54, 'max_bin': 418, 'od_type': 'Iter'}. Best is trial 26 with value: 2.996221780232261.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/relea

0:	learn: 15.5279270	test: 15.5279270	test1: 15.5254762	best: 15.5254762 (0)	total: 1.22s	remaining: 45m 45s
100:	learn: 7.6843165	test: 7.6843165	test1: 7.6791430	best: 7.6791430 (100)	total: 56.1s	remaining: 19m 59s
200:	learn: 7.0358218	test: 7.0358218	test1: 7.0333866	best: 7.0333866 (200)	total: 1m 44s	remaining: 17m 47s
300:	learn: 6.6704942	test: 6.6704942	test1: 6.6722971	best: 6.6722971 (300)	total: 2m 32s	remaining: 16m 30s
400:	learn: 6.3662171	test: 6.3662171	test1: 6.3715411	best: 6.3715411 (400)	total: 3m 20s	remaining: 15m 31s
500:	learn: 5.7443277	test: 5.7443277	test1: 5.7527280	best: 5.7527280 (500)	total: 4m 27s	remaining: 15m 38s
600:	learn: 5.3615060	test: 5.3615060	test1: 5.3714067	best: 5.3714067 (600)	total: 5m 36s	remaining: 15m 29s
700:	learn: 5.1342464	test: 5.1342464	test1: 5.1462746	best: 5.1462746 (700)	total: 6m 45s	remaining: 15m 2s
800:	learn: 4.9767866	test: 4.9767866	test1: 4.9924031	best: 4.9924031 (800)	total: 7m 55s	remaining: 14m 25s
900:	learn: 4

[I 2022-10-24 02:00:56,377] Trial 27 finished with value: 3.1054518116577317 and parameters: {'learning_rate': 0.040637884151287475, 'bagging_temperature': 36.305845955243896, 'n_estimators': 2260, 'max_depth': 15, 'random_strength': 86, 'colsample_bylevel': 0.4033328201915544, 'l2_leaf_reg': 1.6926497913910582e-05, 'min_child_samples': 38, 'max_bin': 381, 'od_type': 'Iter'}. Best is trial 26 with value: 2.996221780232261.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/rele

0:	learn: 15.5079903	test: 15.5079903	test1: 15.5126740	best: 15.5126740 (0)	total: 532ms	remaining: 33m 6s
100:	learn: 7.0358108	test: 7.0358108	test1: 7.0427731	best: 7.0427731 (100)	total: 1m 35s	remaining: 57m 25s
200:	learn: 6.5111126	test: 6.5111126	test1: 6.5222838	best: 6.5222838 (200)	total: 2m 48s	remaining: 49m 21s
300:	learn: 6.2297736	test: 6.2297736	test1: 6.2464050	best: 6.2464050 (300)	total: 4m 5s	remaining: 46m 37s
400:	learn: 5.6413859	test: 5.6413859	test1: 5.6612952	best: 5.6612952 (400)	total: 5m 53s	remaining: 49m 1s
500:	learn: 5.2292961	test: 5.2292961	test1: 5.2516287	best: 5.2516287 (500)	total: 7m 52s	remaining: 50m 54s
600:	learn: 4.9812855	test: 4.9812855	test1: 5.0066424	best: 5.0066424 (600)	total: 9m 59s	remaining: 52m 8s
700:	learn: 4.8283644	test: 4.8283644	test1: 4.8609391	best: 4.8609391 (700)	total: 12m 8s	remaining: 52m 33s
800:	learn: 4.7206827	test: 4.7206827	test1: 4.7604159	best: 4.7604159 (800)	total: 14m 16s	remaining: 52m 17s
900:	learn: 4.

[I 2022-10-24 03:22:31,853] Trial 28 finished with value: 2.9859445101518953 and parameters: {'learning_rate': 0.04814728699783611, 'bagging_temperature': 40.52044595178952, 'n_estimators': 3736, 'max_depth': 16, 'random_strength': 59, 'colsample_bylevel': 0.4861875826015146, 'l2_leaf_reg': 2.309535224342252e-05, 'min_child_samples': 58, 'max_bin': 424, 'od_type': 'Iter'}. Best is trial 28 with value: 2.9859445101518953.
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:4: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  'learning_rate' : trial.suggest_loguniform('learning_rate', 0.01, 0.05),
C:\Users\jkpark\AppData\Local\Temp\ipykernel_14420\354051608.py:5: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releas

0:	learn: 15.5111995	test: 15.5111995	test1: 15.5078792	best: 15.5078792 (0)	total: 1.77s	remaining: 1h 28m
100:	learn: 7.4717051	test: 7.4717051	test1: 7.4820892	best: 7.4820892 (100)	total: 1m 9s	remaining: 33m 4s
200:	learn: 6.8224938	test: 6.8224938	test1: 6.8407885	best: 6.8407885 (200)	total: 2m 18s	remaining: 32m 4s
300:	learn: 6.4777624	test: 6.4777624	test1: 6.5023905	best: 6.5023905 (300)	total: 3m 21s	remaining: 30m 1s
400:	learn: 5.7690004	test: 5.7690004	test1: 5.7973105	best: 5.7973105 (400)	total: 5m 3s	remaining: 32m 36s
500:	learn: 5.2905021	test: 5.2905021	test1: 5.3186723	best: 5.3186723 (500)	total: 6m 52s	remaining: 34m 10s
600:	learn: 5.0260040	test: 5.0260040	test1: 5.0565062	best: 5.0565062 (600)	total: 8m 47s	remaining: 34m 57s
700:	learn: 4.8553439	test: 4.8553439	test1: 4.8928376	best: 4.8928376 (700)	total: 10m 44s	remaining: 35m 5s
800:	learn: 4.7382375	test: 4.7382375	test1: 4.7828505	best: 4.7828505 (800)	total: 12m 41s	remaining: 34m 42s
900:	learn: 4.65

[I 2022-10-24 04:22:18,934] Trial 29 finished with value: 3.0127825927583927 and parameters: {'learning_rate': 0.04875709506907699, 'bagging_temperature': 49.87581533270841, 'n_estimators': 2990, 'max_depth': 16, 'random_strength': 87, 'colsample_bylevel': 0.4492431292043895, 'l2_leaf_reg': 2.2785195712824778e-05, 'min_child_samples': 26, 'max_bin': 395, 'od_type': 'Iter'}. Best is trial 28 with value: 2.9859445101518953.


Best Score: 2.9859445101518953
Best trial {'learning_rate': 0.04814728699783611, 'bagging_temperature': 40.52044595178952, 'n_estimators': 3736, 'max_depth': 16, 'random_strength': 59, 'colsample_bylevel': 0.4861875826015146, 'l2_leaf_reg': 2.309535224342252e-05, 'min_child_samples': 58, 'max_bin': 424, 'od_type': 'Iter'}


Best Score: 2.9859445101518953  
Best trial {'learning_rate': 0.04814728699783611, 'bagging_temperature': 40.52044595178952, 'n_estimators': 3736, 'max_depth': 16, 'random_strength': 59, 'colsample_bylevel': 0.4861875826015146, 'l2_leaf_reg': 2.309535224342252e-05, 'min_child_samples': 58, 'max_bin': 424, 'od_type': 'Iter'}

In [24]:
params = study.best_trial.params
cat_model = CatBoostRegressor(**params).fit(X, y)
y_pred = cat_model.predict(test)

import pandas as pd

sample_submission = pd.read_csv('./jeju_data/sample_submission.csv')
sample_submission['target'] = y_pred
sample_submission.to_csv("./submit_cat_optuna.csv", index=False)

0:	learn: 15.5716201	total: 349ms	remaining: 21m 43s
1:	learn: 15.1267920	total: 1.13s	remaining: 35m 10s
2:	learn: 14.7153248	total: 2.46s	remaining: 51m 4s
3:	learn: 14.3778033	total: 2.72s	remaining: 42m 17s
4:	learn: 14.1375031	total: 2.95s	remaining: 36m 38s
5:	learn: 13.7320558	total: 4.26s	remaining: 44m 8s
6:	learn: 13.3522024	total: 5.72s	remaining: 50m 46s
7:	learn: 12.9751077	total: 7.34s	remaining: 57m 2s
8:	learn: 12.6400785	total: 8.84s	remaining: 1h 1m 2s
9:	learn: 12.3462009	total: 10.5s	remaining: 1h 5m 16s
10:	learn: 12.0403986	total: 12s	remaining: 1h 7m 40s
11:	learn: 11.8349714	total: 12.4s	remaining: 1h 4m
12:	learn: 11.5785669	total: 13.7s	remaining: 1h 5m 24s
13:	learn: 11.3470364	total: 15.3s	remaining: 1h 7m 44s
14:	learn: 11.1201618	total: 16.5s	remaining: 1h 8m 2s
15:	learn: 10.9087378	total: 17.9s	remaining: 1h 9m 12s
16:	learn: 10.7733038	total: 18.2s	remaining: 1h 6m 31s
17:	learn: 10.5773935	total: 19.7s	remaining: 1h 7m 46s
18:	learn: 10.3737512	total: 

In [25]:
import pandas as pd
sample_submission = pd.read_csv('./jeju_data/sample_submission.csv')

params = study.best_trial.params


from sklearn.model_selection import StratifiedKFold
from catboost import CatBoostRegressor
from sklearn.metrics import mean_absolute_error

skf = StratifiedKFold(n_splits=5, shuffle=True, random_state=42)

folds = []

for train_idx, val_idx in skf.split(X, y):
    folds.append((train_idx, val_idx))

CAT_model= {}

for f in range(5):
      print(f'===================================={f+1}============================================')
      train_idx, val_idx = folds[f]
      
      x_train, x_val, y_train, y_val = X.iloc[train_idx], X.iloc[val_idx], y.iloc[train_idx], y.iloc[val_idx]
      
      CAT = CatBoostRegressor(**params)
      CAT.fit(x_train, y_train)
      
      y_pred = CAT.predict(x_val)
      mae = mean_absolute_error(y_val, y_pred)
      print(f"{f + 1} Fold MAE = {mae}")
      CAT_model[f] = CAT
      print(f'================================================================================\n\n')
              

for fold in range(5):
    sample_submission['target'] += CAT_model[fold].predict(test)/5    
sample_submission.to_csv("./catcat.csv", index=False)

In [30]:
a = pd.read_csv('catcat.csv')
b = pd.read_csv('./model/XGB/public 3.14 xgb.csv')

In [32]:
c = (a['target']*0.5) + (b['target']*0.5)
sample_submission = pd.read_csv('./jeju_data/sample_submission.csv')
sample_submission['target'] = c
sample_submission.to_csv("./submit_ff.csv", index=False)